main/commons/src/main/java/org/cryptomator/common/mountpoint/MountPointChooser.java

```diff
@@ -0,0 +1,170 @@
+
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Base interface for the Mountpoint-Choosing-Operation that results in the choice and
+ * preparation of a Mountpoint or an exception otherwise.<br>
+ * <p>All <i>MountPointChoosers (MPCs)</i> need to implement this class and must be added to
+ * the pool of possible MPCs by {@link MountPointChooserModule MountPointChooserModule.}
+ * The MountPointChooserModule requires all {@link dagger.Provides Providermethods} to
+ * be annotated with the {@link PhaseKey @PhaseKey-Annotation} and a <b>unique</b> Phase to
+ * allow the Module to sort them according to the phases' {@link Phase#getTiming()} timing.}
+ * The timing must be defined by the developer to reflect a useful execution order.
+ *
+ * <p><b>Phase-Uniqueness:</b> Phases must be unique, meaning that they must not be used
+ * to annotate more than <i>one</i> Providermethod. Define a new phase for each additional
+ * MPC that is added to the Module. Timings can be reused; the order of Phases with equal
+ * timing among themselves is undefined.
+ *
+ * <p>MPCs are executed by a {@link Volume} in the order of their phase's timing to find
+ * and prepare a suitable Mountpoint for the Volume. The Volume only has access to a {@link Set}
+ * of MPCs in this specific order, that is provided by the Module; the according Phases and exact
+ * timings are inaccessible to the Volume. The Set only contains Choosers that were deemed
+ * {@link #isApplicable() applicable} by the Module.
+ *
+ * <p>At execution of a MPC {@link #chooseMountPoint()} and then {@link #prepare(Path)} are called
+ * by the Volume. If {@code #chooseMountPoint()} yields no result, the next MPC is executed
+ * without first calling the {@code #prepare(Path)}-Method of the current MPC.
+ * This is repeated until<br>
+ * <ul>
+ *     <li><b>either</b> a Mountpoint is returned by {@code #chooseMountPoint()}
+ *     and {@code #prepare(Path)} succeeds or fails</li>
+ *     <li><b>or</b> no MPC remains and an {@link InvalidMountPointException} is thrown.</li>
+ * </ul>
+ * If the {@code #prepare(Path)}-Method of a MPC fails the entire Mountpoint-Choosing-Operation
+ * is aborted and the method should do all necessary cleanup before throwing the exception.
+ * If the preparation succeeds {@link #cleanup(Path)} can be used after unmount to do any
+ * remaining cleanup.
+ */
+public interface MountPointChooser {
+
+	/**
+	 * Called by the {@link MountPointChooserModule} to determine whether this MountPointChooser is
+	 * applicable for the given Systemconfiguration.
+	 *
+	 * <p>The result of this method defaults to true. Developers should override this method to
+	 * check for Systemconfigurations that are unsuitable for this MPC.
+	 *
+	 * @return a boolean flag; true if applicable, else false.
+	 * @see #chooseMountPoint()
+	 */
+	default boolean isApplicable() {
+		return true; //Usually most of the choosers should be applicable
```

Comment:
"Usually" is not enough to define a default method. Since e.g. drive letters are Windows-specific, I prefer to have this method be implemented in _each_ MPC explicitly.

Reply:
Resolved by 5dac594cc922d701f0147da5672335813edd6324